rm102/gui.py

```diff
@@ -50,11 +51,30 @@ def update_gui(self, reg_count, acc, error):
             self.commandRegisterList.model.item(reg_count).setBackground(
                     QBrush(QColorConstants.Red if error else QColorConstants.Red))
 
+    def register_changed(self):
+        self.file_is_open(self.filepath, False)
+
     def runButtonClicked(self):
         self.commandRegisterList.model.run(self.regListModel)
         if self.commandRegisterList.model.stop is not None:
             self.runButton.setEnabled(False)
 
+    def closeEvent(self, event):
+        if not self.changes_saved:
+            filename = filename = self.filepath.split('/')[-1] if self.filepath else 'Unbenannt'
+            msg = QMessageBox()
+            msg.setIcon(QMessageBox.Warning)
+            msg.setWindowTitle("Änderungen speichern?")
+            msg.setText(f"Sollen die Änderungen an {filename} gespeichert werden?")
+            msg.setStandardButtons(QMessageBox.Save | QMessageBox.Discard | QMessageBox.Cancel)
+            msg.setEscapeButton(QMessageBox.Cancel)
+            result = msg.exec()
+
+            if result == QMessageBox.Cancel:
+                event.ignore()
+            if result == QMessageBox.Save:
+                self.saveButtonClicked()
```

Comment:
For the framework's convinience, there should be `event.accept()`

Reply:
Solved in 4a8511e22ed1ab409cc91e0451d1582b76a42357